examples/e-commerce/src/app/app.component.ts

```diff
@@ -1,14 +1,38 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import algoliasearch from 'algoliasearch/lite';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
   config = {
     searchClient: algoliasearch('latency', '6be0576ff61c053d5f9a3225e2a90f76'),
     indexName: 'instant_search',
     routing: true,
   };
+  resultsContainer = undefined;
+
+  onKeyUp = event => {
+    if (event.key !== 'Escape') {
+      return;
+    }
+    this.closeFilters();
+  };
+
+  ngOnInit() {
+    this.resultsContainer = document.querySelector('.container-results');
```

Comment:
To my knowledge in Angular the safest way to access a dom element is in ngAfterViewInit lifecycle method, and it comes after ngOnInit (see full [component lifecycle](https://angular.io/guide/lifecycle-hooks)).

Other than that, just FYI, the angulary way for accessing elements is using ViewChild and ElementRef.
https://alligator.io/angular/viewchild-access-component/#dom-elements
But if this works for I guess just keep it :)

Reply:
Thanks!
I knew about ViewChild but forgot about it when doing this. Anyway because `.container-results` is a class name but not an id, I should stick to the current implementation (querySelector) for this.
I've replaced OnInint with AfterViewInit.

Reply:
209d19c